jobsdb/migration.go

```diff
@@ -153,6 +153,52 @@ func (jd *HandleT) doMigrateDS(ctx context.Context) error {
 	return err
 }
 
+// based on size of given DSs, gives a list of DSs for us to vaccum status tables
+func (jd *HandleT) getVacuumCandidates(ctx context.Context, dsList []dataSetT) ([]dataSetT, error) {
+	// get name and it's size of all tables
+	var rows *sql.Rows
+	rows, err := jd.dbHandle.QueryContext(
+		ctx,
+		`SELECT pg_total_relation_size(oid) AS size, relname
+		FROM pg_class
+		where relname = ANY(
+			SELECT tablename
+				FROM pg_catalog.pg_tables
+				WHERE schemaname NOT IN ('pg_catalog','information_schema')
+				AND tablename like $1
+		)`,
+		jd.tablePrefix+"_job%",
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = rows.Close() }()
+
+	tableSizes := map[string]int64{}
+	for rows.Next() {
+		var (
+			tableSize int64
+			tableName string
+		)
+		err = rows.Scan(&tableSize, &tableName)
+		if err != nil {
+			return nil, err
+		}
+		tableSizes[tableName] = tableSize
+	}
+	err = rows.Err()
+	if err != nil {
+		return nil, err
+	}
+
+	datasets := lo.Filter(dsList,
```

Comment:
True, but the query gives me all the tables and their sizes. I'm doing the filtering since we are only interested in a few status tables that are going to be cleaned

Reply:
I meant something like this:

```
    .
    .
    .
	defer func() { _ = rows.Close() }()
    toVacuum := map[string]struct{}{}
	for rows.Next() {
		var (
			tableSize int64
			tableName string
		)
		err = rows.Scan(&tableSize, &tableName)
		if err != nil {
			return nil, err
		}
        if tableSize > vacuumFullStatusTableThreshold {
				toVacuum[tableName] = struct{}{}
		}
	}
	err = rows.Err()
	if err != nil {
		return nil, err
	}
    return toVacuum, nil
}
```

Why do we need to pass `dsList` to this function..?

Reply:
I just read your comment again and understand now that you were wanting to vacuum only tables that were marked for compaction.
Might as well vacuum all..?

Reply:
or if you intend to vacuum only a subset of `dsList`, you may use it as part of the query.

Reply:
I think we can vacuum all. WDYT @atzoum ?